### PR TITLE
allow  https:// in proxy parameter in tsh

### LINKED ISSUE #27030: Support https:// for proxy server references
What would you like Teleport to do?

Teleport Connect always you to specify the proxy server with the `https://` prefix.  If you specify that in CLI or configs you get the below type of error. 

```bash
tsh login --proxy=https://teleport.example.com
ERROR: Get "https://https//teleport.example.com/webapi/ping": dial tcp: lookup https: no such host
```

### PR DESCRIPTION
**Pull request by stevenGravy**
fixes https://github.com/gravitational/teleport/issues/27030

### CODE REVIEW

File: tool/tsh/common/tsh.go [hunk 1]
```diff
@@ -1115,6 +1115,10 @@ func Run(ctx context.Context, args []string, opts ...CliOption) error {
 		}
 	}
 
+	// Remove HTTPS:// in proxy parameter as this is automatically handled
+	var httpsRE = regexp.MustCompile(`(?mi)https://`)
```
Comment (zmb3) [inline tool/tsh/common/tsh.go:1119]: Regex seems overkill here. Why not just a `TrimPrefix`?

Comment (stevenGravy) [reply tool/tsh/common/tsh.go:1119]: that would miss `HTTPS://` and just get `https://`. If we think that's a low enough scenario sure I can use TrimPrefix.

Comment (stevenGravy) [reply tool/tsh/common/tsh.go]: @zmb3 converted to `TrimPrefix` of `https://` and `HTTPS://` which should cover most scenarios.

File: tool/tsh/common/tsh_test.go [hunk 2]
```diff
@@ -691,6 +691,45 @@ func TestRelogin(t *testing.T) {
 	require.NoError(t, err)
 }
 
+// Test when https:// is included in --proxy address
+func TestIgnoreHttpsprefix(t *testing.T) {
```
Comment (zmb3) [inline tool/tsh/common/tsh_test.go:695]:
```suggestion
func TestIgnoreHTTPSPrefix(t *testing.T) {
```

File: tool/tsh/common/tsh_test.go [hunk 3]
```diff
@@ -691,6 +691,45 @@ func TestRelogin(t *testing.T) {
...
+		withBootstrap(connector, alice),
+	)
+
+	authServer := authProcess.GetAuthServer()
+	require.NotNil(t, authServer)
+
+	proxyAddr, err := proxyProcess.ProxyWebAddr()
+	require.NoError(t, err)
+
+	buf := bytes.NewBuffer([]byte{})
```
Comment (zmb3) [inline tool/tsh/common/tsh_test.go:716]:
```suggestion
	var buf bytes.Buffer
```

File: tool/tsh/common/tsh_test.go [hunk 4]
```diff
@@ -691,6 +691,45 @@ func TestRelogin(t *testing.T) {
...
+	proxyAddress := "https://" + proxyAddr.String()
+	err = Run(context.Background(), []string{
+		"login",
+		"--insecure",
+		"--debug",
+		"--auth", connector.GetName(),
+		"--proxy", proxyAddress,
+	}, setHomePath(tmpHomePath), func(cf *CLIConf) error {
+		cf.MockSSOLogin = mockSSOLogin(t, authServer, alice)
+		cf.overrideStderr = buf
```
Comment (zmb3) [inline tool/tsh/common/tsh_test.go:727]:
```suggestion
		cf.overrideStderr = &buf
```

If taken with the suggestion above

**Review by zmb3 — APPROVED** [review]

**Review by xacrimon — APPROVED** [review]

**Pull request merged**